### DISCUSSION TITLE
Only item alias in sef URL

### DISCUSSION THREAD

**Discussion by laurentgarin** [joomengine/Joomla-Component-Builder#628] [Q&A] [answered]
Hello,

Would it be possible when SEF is enabled, instead of having URLs such as :
 http://www.wesite.com/menu/viewname/itemalias
like this rather http://www.wesite.com/itemalias

In fact keep only the alias of the item.

I think it is of course related to the router, how could I do it?

Thank you for your help.

**Comment by Llewellynvdm** [comment, marked as answer]
This is not an automatic feature of Joomla, you need to link the item to a menu for this kind of SEO. Since Joomla has no way of knowing what component needs to be queried with just one word, unless it is linked to a menu.

**Reply by laurentgarin** [reply]
OK thanks, do you think this will be possible with a plugin system perhaps ?

**Reply by Llewellynvdm** [reply]
This is a plugin system... the SEF system, so you can look at it here: https://github.com/joomla/joomla-cms/blob/3.9.23/plugins/system/sef/sef.php

But you need to think about the fact that there is two things happening all the time:
convert from `index.php?option=com_component&view=viewname&task=dothis&id=23` => `component/view/alias`
convert from `component/view/alias` => `index.php?option=com_component&view=viewname&task=dothis&id=23`

Since we need the `index.php?option=com_component&view=viewname&task=dothis&id=23` values to know what to do, and we need the `component/view/alias` as SEF. This conversion back and forth is happening on every page load, when any link is created, and clicked to another internal page.

We then have a very basic URL, it can get far more advanced with far more pieces to take into account. Like the following:
`index.php?option=com_component&view=viewname&task=dothis&type=5&cid=21&id=23`

This complicates the task, and is why we have allowed custom coding in the dynamicGet areas that target the router.

Therefore we can on menu level do much better, this means we can have a profile menu, and have it link to each user directly. Checkout this project, where I am making use of this: https://coral-demo.ban.com.na/demo

Login as a staff, and then search for a user... open this user and watch the URL you will see it maps to the `dashboard/profile/useralias` now the `profile` is a menu item, and the `useralias` is dynamic so we could make it `profile/useralias` if we wanted to. To get this behavior you need to look at how SD is build to manage its menu items, what [these tutorials](https://www.youtube.com/c/JoomlaMount/search?query=menu) if you have not already.
